django_pg_simple_hll/aggregate.py

```diff
@@ -0,0 +1,7 @@
+from django.db.models import Aggregate, IntegerField
+
+
+class ApproxCardinality(Aggregate):
```

Comment:
I've added some of the other properties like Count. 

The init function is probably not relevant. The aggregation wont work on `*` input, only on a specific column.

In terms of language, I'm not sure that I would want to change it to Count. The algorithm is specific for approximating cardinality – the number of distinct elements in a set – not counting them.

Reply:
Yea I'd be inclined to stick to postgres language though really since the name is more for the technical user who shouldn't really need to be worrying about the exact hyperloglog implementation or set theory.

There also seems to be some ambiguity about the definition of cardinality of a multiset. Wiki seems to think that actually doesn't mean the distinct count at all:

> The cardinality of a multiset is the sum of the multiplicities of all its elements. For example, in the multiset {a, a, b, b, b, c} the multiplicities of the members a, b, and c are respectively 2, 3, and 1, and therefore the cardinality of this multiset is 6.

Whereas other sources seem to think the cardinality there would be 3.

I think `DistinctCount` is probably a bit clearer and more in plain English

Reply:
> The algorithm is specific for approximating cardinality – the number of distinct elements in a set – not counting them.

Somewhat along the lines of what Alex has said, I'm not sure this is quite what cardinality is, at least in the classic mathematical setting.  

The motivation for cardinality was not to have an alternative term for distinct counts, but to able extend the concept of "size" in set theory from a count (which would be distinct anyway) of finite sets to a measure of infinite sets.  I don't think "distinctness" is really vital to the idea of cardinality -  it's just implicit because cardinality is associated with sets.  Since we're not dealing with sets here anyway I think it probably is worth being explicit about the distinctness.

I don't actually have a strong opinion on the name, I think it's fairly obvious what you mean. I just wanted to get invovled in the chat. Cantor's proofs got me into mathematics.

Reply:
I should have known not to explain terminology with mathematicians.

However, I don't think that I am making a wild claim here. It looks like cardinality is the common language used for this exact operation in databases: 
postgres: https://github.com/citusdata/postgresql-hll 
redis: https://redis.io/docs/data-types/hyperloglogs/ 
elasticsearch: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-cardinality-aggregation.html 

And it is the way that it is called in the original research papers:
http://algo.inria.fr/flajolet/Publications/FlMa85.pdf
http://algo.inria.fr/flajolet/Publications/FlFuGaMe07.pdf

Reply:
I think cardinality is fine if you like long words over count but I'd still say the distinct bit needs to be explicit unless you're going to explicitly reference in the name that this uses the HLL algorithm.

Reply:
My 2cents on the name is that something like `ApproximateCount` is easier for devs not familiar with HLL to understand, however if you don't understand the drawbacks of of this aggregation then you probably shouldn't be using it? Also this aggregate is very tightly coupled to the HLL implementation (eg. we wouldn't swap this out with [another approximate count algorithm](https://web.archive.org/web/20230321011643/https://wiki.postgresql.org/wiki/Count_estimate)) so I think it's fine if it uses language more associated with HLL.

Maybe a docstring could be a good compromise to help with IDEs if someone comes across this for the first time in a codebase rather than viewing this source repo?
```suggestion
class ApproxCardinality(Aggregate):
    """
    Return an approximate count based on the HyperLogLog algorithm.
    
    This should be much faster than Django's Count but will give
    an approximate result rather than an exact count.
    """
```

Reply:
> > The aggregation wont work on `*` input, only on a specific column
> 
> By "won't work" do you mean Django/Posgres will raise an exception or it will give inaccurate/nonsensical results? If it's the latter then we should catch that to prevent bad data being returned

It will raise errors. I've added tests here 59ab35771e4abafca36ac7b1fd26873f2c0fd2ab